docs/SPECIFICATION.md

```diff
@@ -0,0 +1,703 @@
+# Onyxia API Specification
+<p style="text-align: right;"> (back to <a href='./README.md'>Onyxia API Documentation</a>) <p>
+
+
+## Table of Contents
+1. [Terminology](#terminology)
+2. [Endpoints](#endpoints)
```

Comment:
I don't think throughoutly documenting the endpoints here, in a static documentation is the best way to do. It will eventually become a liability having to update this each time we change things in any endpoint.  
I think we should instead double down on the "swagger" (openapi) way and use it as the single source of truth. There are many improvements that can be done on the current usage of swagger (fixing the reverse proxy redirection, documenting the endpoints and inputs, adding a link to it in this documentation ...) but I really think we want to stick to a single source of truth for this

Reply:
Agree that the only source of truth should be the Open API Specification, cleaned up the exhaustive documentation here fb1eea47f0c71f0574407dddd25fa81602ba32f9, however, left some information for a first approach to the API - so that a new reader can have a notion of what the API entails. Will comeback with additions on the OAS

Reply:
Main endpoints documented in OAS:
- 1bd3caf5cde630925ba34841b0dca9a407ff07dd for /public endpoints
- f3ce198d7fcc4c26d7b8f0a495b2a5a4baacf81f for /my-lab endpoints
- ed468e8a82260c0c6e67ddb2a4be43385a7723de for /my-lab/quotas endpoints (was this 789485e2d3e92a2888a9f8b78918db4aecb7feb6 the desired output in the /my-lab/quotas/reset ?)